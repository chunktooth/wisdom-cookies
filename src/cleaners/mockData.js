/* exported variableName */
export const mockQuotes = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];