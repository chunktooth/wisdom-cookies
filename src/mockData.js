/* exported variableName */
export const mockWisdoms = [
  {
    'message': 'Mankind is unkind, man..',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeeeed!',
    'id': '1234'
  }
];

export const mockJar = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];

export const mockMoonphases = { response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
};

export const mockMoonphasesArray = [{ response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
}]; 

export const mockMoonphase =  {
  date: "2018-04-22", 
  name: "first quarter"
};

export const mockWisdom = {
  'message': 'In Crust We Trust',
  'id': '1234'
};

export const mockCleanedMoons = [
  {
    date: "2018-04-22", 
    name: "first quarter"
  },
  { 
    date: "2018-04-30", 
    name: "full moon"
  }
];