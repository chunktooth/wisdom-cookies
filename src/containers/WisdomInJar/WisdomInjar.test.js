import React from 'react';
import { WisdomInJar, 
  mapStateToProps,
  mapDispatchToProps } from './WisdomInJar';
import { shallow } from 'enzyme';
import { mockWisdom, mockJar } from '../../mockData';

describe('WisdomInJar', () => {
  let wrapper;
  const mockTrashFromJar = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<WisdomInJar 
      jar={mockJar}
      wisdom={mockWisdom}
      trashFromJar={mockTrashFromJar} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call trashFromJar when calling removeFromJar', () => {
    wrapper = shallow(<WisdomInJar 
      jar={mockJar}
      wisdom={mockWisdom}
      trashFromJar={mockTrashFromJar} />);
    
    wrapper.instance().removeFromJar();
    expect(mockTrashFromJar).toHaveBeenCalledWith(mockWisdom.id);
  });

  it('should map wisdom state to props', () => {
    const mappedWisdom = mapStateToProps(mockWisdom);
    expect(mappedWisdom.wisdom).toEqual(mockWisdom.wisdom);
  });

  it('should map jar state to props', () => {
    const mappedJar = mapStateToProps(mockJar);
    expect(mappedJar.jar).toEqual(mockJar.jar);
  });

  it('should dispatch trashFromJar to props', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.trashFromJar();
    expect(mockDispatch).toHaveBeenCalled();
  });

});