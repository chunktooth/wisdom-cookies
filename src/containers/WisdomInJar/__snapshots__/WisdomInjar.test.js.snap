// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WisdomInJar should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WisdomInJar
    jar={
        Array [
            Object {
              "id": "5280",
              "message": "Mankind is unkind, man",
            },
            Object {
              "id": "1234",
              "message": "Godluck and Goodspeed!",
            },
          ]
    }
    trashFromJar={[Function]}
    wisdom={
        Object {
            "id": "1234",
            "message": "In Crust We Trust",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="delete-btn"
          onClick={[Function]}
>
          [ X ]
</button>,
        <p
          className="wisdom-message"
>
          In Crust We Trust
</p>,
      ],
      "className": "WisdomInJar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "[ X ]",
          "className": "delete-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "[ X ]",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "In Crust We Trust",
          "className": "wisdom-message",
        },
        "ref": null,
        "rendered": "In Crust We Trust",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="delete-btn"
            onClick={[Function]}
>
            [ X ]
</button>,
          <p
            className="wisdom-message"
>
            In Crust We Trust
</p>,
        ],
        "className": "WisdomInJar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "[ X ]",
            "className": "delete-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "[ X ]",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "In Crust We Trust",
            "className": "wisdom-message",
          },
          "ref": null,
          "rendered": "In Crust We Trust",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
